ouroboros-network-protocols/src/Ouroboros/Network/Protocol/ChainSync/PipeClient.hs

```diff
@@ -0,0 +1,70 @@
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE GADTs               #-}
+{-# LANGUAGE NamedFieldPuns      #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE RankNTypes #-}
+
+module Ouroboros.Network.Protocol.ChainSync.PipeClient
+where
+
+import Network.TypedProtocol
+import Ouroboros.Network.Protocol.ChainSync.Client
+import Ouroboros.Network.Protocol.ChainSync.Type
+import Pipes
+
+data ChainSyncResponse header point tip
+  = RollForward header tip
+  | RollBackward point tip
+  | IntersectFound point tip
+  | IntersectNotFound tip
```

Comment:
This looks like a good start but I think where we want to go next is to not include the intersect finding in the pipe event stream, but to do all the intersection finding first and then yield the pipe producer (if the intersection finding is successful) for streaming the blocks.

So in the end we want just the roll forwards and roll backwards as pipe events.

And later we might want to rename it something like:
```
data BlockOrRollback block = NextBlock block | Rollback Point
```